pkg/occlient/occlient.go

```diff
@@ -201,99 +203,10 @@ func imageWithMetadata(image *imagev1.Image) error {
 	return nil
 }
 
-// getOcBinary returns full path to oc binary
-// first it looks for env variable KUBECTL_PLUGINS_CALLER (run as oc plugin)
-// than looks for env variable OC_BIN (set manualy by user)
-// at last it tries to find oc in default PATH
-func getOcBinary() (string, error) {
-	log.Debug("getOcBinary - searching for oc binary")
-
-	var ocPath string
-
-	envKubectlPluginCaller := os.Getenv("KUBECTL_PLUGINS_CALLER")
-	envOcBin := os.Getenv("OC_BIN")
-
-	log.Debugf("envKubectlPluginCaller = %s\n", envKubectlPluginCaller)
-	log.Debugf("envOcBin = %s\n", envOcBin)
-
-	if len(envKubectlPluginCaller) > 0 {
-		log.Debug("using path from KUBECTL_PLUGINS_CALLER")
-		ocPath = envKubectlPluginCaller
-	} else if len(envOcBin) > 0 {
-		log.Debug("using path from OC_BIN")
-		ocPath = envOcBin
-	} else {
-		path, err := exec.LookPath("oc")
-		if err != nil {
-			log.Debug("oc binary not found in PATH")
-			return "", err
-		}
-		log.Debug("using oc from PATH")
-		ocPath = path
-	}
-	log.Debug("using oc from %s", ocPath)
-
-	if _, err := os.Stat(ocPath); err != nil {
-		return "", err
-	}
-
-	return ocPath, nil
-}
-
-type OcCommand struct {
-	args   []string
-	data   *string
-	format string
-}
-
-// runOcCommands executes oc
-// args - command line arguments to be passed to oc ('-o json' is added by default if data is not nil)
-// data - is a pointer to a string, if set than data is given to command to stdin ('-f -' is added to args as default)
-func (c *Client) runOcComamnd(command *OcCommand) ([]byte, error) {
-	cmd := exec.Command(c.ocpath, command.args...)
-
-	// if data is not set assume that it is get command
-	if len(command.format) > 0 {
-		cmd.Args = append(cmd.Args, "-o", command.format)
-	}
-	if command.data != nil {
-		// data is given, assume this is create or apply command
-		// that takes data from stdin
-		cmd.Args = append(cmd.Args, "-f", "-")
-
-		// Read from stdin
-		stdin, err := cmd.StdinPipe()
-		if err != nil {
-			return nil, err
-		}
-
-		// Write to stdin
-		go func() {
-			defer stdin.Close()
-			_, err := io.WriteString(stdin, *command.data)
-			if err != nil {
-				fmt.Printf("can't write to stdin %v\n", err)
-			}
-		}()
-	}
-
-	log.Debugf("running oc command with arguments: %s\n", strings.Join(cmd.Args, " "))
-
-	output, err := cmd.CombinedOutput()
-	if err != nil {
-		if _, ok := err.(*exec.ExitError); ok {
-			return nil, errors.Wrapf(err, "command: %v failed to run:\n%v", cmd.Args, string(output))
-		}
-		return nil, errors.Wrap(err, "unable to get combined output")
-	}
-
-	return output, nil
-}
-
-func isLoggedIn(ocpath string) bool {
-	cmd := exec.Command(ocpath, "whoami")
-	output, err := cmd.CombinedOutput()
-	log.Debugf("isLoggedIn err:  %#v \n output: %#v", err, string(output))
+// isLoggedIn checks whether user is logged in or not and returns boolean output
+func (c *Client) isLoggedIn() bool {
+	output, err := c.userClient.Users().Get("~", metav1.GetOptions{})
```

Comment:
What's the purpose of the constant "~"? May have to add a comment here (since I don't understand either)

Reply:
Home maybe?